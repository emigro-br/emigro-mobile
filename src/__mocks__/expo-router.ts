import { ExpoRouter } from 'expo-router/types/expo-router';

const routerMock = {
  back: jest.fn(),
  canGoBack: jest.fn(),
  push: jest.fn(),
  navigate: jest.fn(),
  replace: jest.fn(),
  dismiss: jest.fn(),
  dismissAll: jest.fn(),
  canDismiss: jest.fn(),
  setParams: jest.fn(),
};

export const useRouter = (): ExpoRouter.Router => routerMock;

export const useLocalSearchParams = jest.fn().mockImplementation(() => ({}));

export const useFocusEffect = jest.fn();

const stackMock = {
  Screen: jest.fn(),
  Navigator: jest.fn(),
};

export const Stack = stackMock;
