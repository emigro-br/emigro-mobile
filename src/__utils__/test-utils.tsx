import { JSX, JSXElementConstructor, ReactElement } from 'react';

import { RenderOptions, render } from '@testing-library/react-native';

import { ThemeProvider } from './ThemeProvider';

type Props = {
  children: JSX.Element;
};

const AllTheProviders = ({ children }: Props) => {
  return <ThemeProvider theme="light">{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement<unknown, string | JSXElementConstructor<any>>,
  options?: RenderOptions | undefined,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
