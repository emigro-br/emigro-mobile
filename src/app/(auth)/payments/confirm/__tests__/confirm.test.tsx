import React from 'react';

import { fireEvent, waitFor } from '@testing-library/react-native';
import { useRouter } from 'expo-router';

import { inputPIN, render } from 'test-utils';

import * as quotesService from '@/services/emigro/quotes';
import { Transaction } from '@/services/emigro/types';
import { balanceStore } from '@/stores/BalanceStore';
import { paymentStore } from '@/stores/PaymentStore';
import { Payment, PixPayment } from '@/types/PixPayment';
import { CryptoAsset } from '@/types/assets';

import { ConfirmPayment } from '..';

jest.mock('@/stores/SessionStore', () => ({
  sessionStore: {
    publicKey: 'mockPublicKey',
  },
}));

jest.mock('@/stores/SecurityStore', () => ({
  securityStore: {
    pin: '1234',
    verifyPin: jest.fn().mockResolvedValue(true),
  },
}));

jest.mock('@/stores/BalanceStore', () => ({
  balanceStore: {
    get: jest.fn().mockReturnValue(100),
    currentAssets: jest.fn().mockReturnValue([]),
  },
}));

describe('ConfirmPayment component', () => {
  const mockScannedPayment: Payment = {
    brCode: 'mocked-brCode',
    assetCode: CryptoAsset.USDC,
    transactionAmount: 10,
    merchantName: 'John Doe',
    merchantCity: '123 Main St',
    walletKey: 'mocked-walletkey',
  };

  const mockPixPayment: PixPayment = {
    assetCode: CryptoAsset.USDC,
    transactionAmount: 10,
    merchantName: 'John Doe',
    merchantCity: '123 Main St',
    pixKey: 'mocked-pixkey',
    // pix fields
    brCode: 'mocked-brCode',
    txid: 'mocked-identifier',
    taxId: 'mocked-taxId',
    bankName: 'Mock Bank',
  };

  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();
    paymentStore.setScannedPayment(mockScannedPayment);
  });

  it('renders correctly for stellar payment', async () => {
    paymentStore.setScannedPayment(mockScannedPayment); // for full test coverage
    jest.spyOn(quotesService, 'fetchQuote');
    // const { getByText, queryByText, queryAllByText } = render(<ConfirmPayment />);
    const screen = render(<ConfirmPayment />);

    // receiver info
    // expect(getByText('Review the payment')).toBeOnTheScreen();
    // expect(queryAllByText('$ 10.00')).toHaveLength(2); // the amount is displayed twice
    // expect(getByText('for John Doe')).toBeOnTheScreen();
    // expect(getByText('in 123 Main St')).toBeOnTheScreen();

    // // stellar info
    // expect(getByText('Institution:')).toBeOnTheScreen();
    // expect(getByText('Stellar Network')).toBeOnTheScreen();
    // expect(getByText('Wallet Key:')).toBeOnTheScreen();
    // expect(getByText('mocke...etkey')).toBeOnTheScreen(); // masked
    // // expect(getByText('Identifier:')).toBeOnTheScreen();

    // // not pix info
    // expect(queryByText('CPF/CNPJ:')).toBeNull();

    // // payment
    // expect(getByText('Select the account')).toBeOnTheScreen();
    // expect(getByText('Balance: $ 100.00')).toBeOnTheScreen();
    // expect(getByText('Pay')).toBeOnTheScreen();

    expect(screen.toJSON()).toMatchSnapshot();

    expect(quotesService.fetchQuote).not.toHaveBeenCalled();
  });

  it('renders correctly for pix payment', async () => {
    paymentStore.setScannedPayment(mockPixPayment); // for full test coverage
    jest.spyOn(quotesService, 'fetchQuote');
    // const { getByText, queryByText, queryAllByText } = render(<ConfirmPayment />);
    const screen = render(<ConfirmPayment />);

    // receiver info
    // expect(getByText('Review the payment')).toBeOnTheScreen();
    // expect(queryAllByText('$ 10.00')).toHaveLength(2); // the amount is displayed twice
    // expect(getByText('for John Doe')).toBeOnTheScreen();
    // expect(getByText('in 123 Main St')).toBeOnTheScreen();

    // // pix info
    // expect(getByText('Pix Key:')).toBeOnTheScreen();
    // expect(getByText('mocked-pixkey')).toBeOnTheScreen();
    // expect(getByText('Identifier:')).toBeOnTheScreen();
    // expect(getByText('mocked-identifier')).toBeOnTheScreen();
    // expect(getByText('CPF/CNPJ:')).toBeOnTheScreen();
    // expect(getByText('mocked-taxId')).toBeOnTheScreen();
    // expect(getByText('Institution:')).toBeOnTheScreen();
    // expect(getByText('Mock Bank')).toBeOnTheScreen();

    // // not stellar
    // expect(queryByText('Wallet Key:')).toBeNull();

    // // payment
    // expect(getByText('Select the account')).toBeOnTheScreen();
    // expect(getByText('Balance: $ 100.00')).toBeOnTheScreen();
    // expect(getByText('Pay')).toBeOnTheScreen();

    expect(screen.toJSON()).toMatchSnapshot();
    expect(quotesService.fetchQuote).not.toHaveBeenCalled();
  });

  it('should quote when change the asset', async () => {
    jest
      .spyOn(quotesService, 'fetchQuote')
      .mockResolvedValueOnce({ source_amount: 99 } as quotesService.IQuoteResponse);
    jest.spyOn(balanceStore, 'get').mockReturnValue(50); // enough balance

    const { getByTestId } = render(<ConfirmPayment />);
    const select = getByTestId('select-account');
    fireEvent(select, 'onChange', { value: 'BRZ' });

    await waitFor(() => {
      expect(getByTestId('balance')).toHaveTextContent('Balance: R$ 50.00');
    });

    await waitFor(() => {
      expect(quotesService.fetchQuote).toHaveBeenCalledWith({
        from: CryptoAsset.BRZ,
        to: CryptoAsset.USDC,
        amount: mockPixPayment.transactionAmount.toString(),
        type: 'strict_receive',
      });
    });
    expect(getByTestId('quote')).toHaveTextContent('R$ 99.00');
  });

  it('should open the edit amount when the "edit" is pressed', async () => {
    const mockPixZeroAmount = { ...mockPixPayment, transactionAmount: 0 };
    paymentStore.setScannedPayment(mockPixZeroAmount); // for full test coverage
    const { getByText, getByTestId } = render(<ConfirmPayment />);
    expect(getByTestId('amount')).toHaveTextContent('$ 0.00');
    const edit = getByText('Edit');
    expect(edit).toBeOnTheScreen();

    fireEvent.press(edit);

    await waitFor(() => {
      expect(getByTestId('input-amount-action-sheet')).toBeOnTheScreen();
    });
  });

  it('calls the handlePressPay function when the "Pay" button is pressed', async () => {
    const setTransactionMock = jest.spyOn(paymentStore, 'setTransaction');
    const { getByText, getByTestId } = render(<ConfirmPayment />);

    const payButton = getByText('Pay');
    fireEvent.press(payButton);

    expect(setTransactionMock).toHaveBeenCalled();
    await waitFor(() => {
      expect(getByTestId('pin-screen')).toBeOnTheScreen();
    });
  });

  it('goes to success screen when the PIN is verified and payment was successfully', async () => {
    const router = useRouter();
    const payMock = jest.spyOn(paymentStore, 'pay').mockResolvedValue({ status: 'paid' } as Transaction);
    const { getByText, getByTestId } = render(<ConfirmPayment />);

    const payButton = getByText('Pay');
    fireEvent.press(payButton);

    await waitFor(() => {
      expect(getByTestId('pin-screen')).toBeOnTheScreen();
    });

    inputPIN('1234');

    await waitFor(() => {
      expect(router.replace).toHaveBeenCalledWith('/<path>/success');
      expect(payMock).toHaveBeenCalled();
    });
  });

  it('goes to waiting screen when the payment is processing', async () => {
    const router = useRouter();
    const payMock = jest.spyOn(paymentStore, 'pay').mockResolvedValue({ status: 'pending' } as Transaction);
    const { getByText, getByTestId } = render(<ConfirmPayment />);

    const payButton = getByText('Pay');
    fireEvent.press(payButton);

    await waitFor(() => {
      expect(getByTestId('pin-screen')).toBeOnTheScreen();
    });

    inputPIN('1234');

    await waitFor(() => {
      expect(router.replace).toHaveBeenCalledWith('/<path>/waiting');
      expect(payMock).toHaveBeenCalled();
    });
  });

  it('goes to error screen when the payment fails', async () => {
    const router = useRouter();
    const payMock = jest.spyOn(paymentStore, 'pay').mockRejectedValue(new Error('Payment failed'));
    const { getByText, getByTestId } = render(<ConfirmPayment />);

    const payButton = getByText('Pay');
    fireEvent.press(payButton);

    await waitFor(() => {
      expect(getByTestId('pin-screen')).toBeOnTheScreen();
    });

    inputPIN('1234');

    await waitFor(() => {
      expect(router.replace).toHaveBeenCalledWith({
        pathname: '/<path>/error',
        params: { error: 'Payment failed' },
      });
      expect(payMock).toHaveBeenCalled();
    });
  });
});
