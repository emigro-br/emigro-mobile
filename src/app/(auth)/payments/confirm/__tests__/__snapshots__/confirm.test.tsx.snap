// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPayment component renders correctly for pix payment 1`] = `
<View
  style={
    [
      {},
      {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    alwaysBounceVertical={true}
    onContentSizeChange={null}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminationRequest={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollShouldSetResponder={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    pagingEnabled={false}
    scrollViewRef={null}
    sendMomentumEvents={false}
    snapToEnd={true}
    snapToStart={true}
    style={
      {
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      }
    }
  >
    <RCTScrollContentView
      collapsable={false}
      style={
        [
          false,
          undefined,
        ]
      }
    >
      <View
        className="flex-1"
        style={
          {
            "paddingTop": 0,
          }
        }
      >
        <View
          className="flex-col gap-4 p-4"
        >
          <View
            className="flex-row justify-between"
          >
            <RCTText
              accessibilityRole="header"
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              isHighlighted={false}
              selectionColor={null}
              style={
                {
                  "fontSize": 37.44,
                  "fontWeight": "bold",
                  "marginVertical": 32,
                }
              }
            >
              Review the payment
            </RCTText>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              dataSet={
                {
                  "active": "false",
                  "focus": "false",
                  "focusVisible": "false",
                  "hover": "false",
                }
              }
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              states={
                {
                  "active": false,
                  "focus": false,
                  "focusVisible": false,
                  "hover": false,
                }
              }
              testID="close-button"
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                meetOrSlice={0}
                minX={0}
                minY={0}
                role="img"
                stroke="currentColor"
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGPath
                      d="M18 6L6 18"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    />
                    <RNSVGPath
                      d="M6 6L18 18"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
          <View
            className="flex-row items-center"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              dataSet={
                {
                  "active": "false",
                  "disabled": "false",
                  "focus": "false",
                  "focusVisible": "false",
                  "hover": "false",
                }
              }
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <RCTText
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                isHighlighted={false}
                selectionColor={null}
                testID="amount"
              >
                $ 10.00
              </RCTText>
            </View>
          </View>
          <View
            className="flex-col gap-7"
          >
            <View>
              <RCTText
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                isHighlighted={false}
                numberOfLines={1}
                selectionColor={null}
              >
                for
                 
                <RCTText
                  isHighlighted={false}
                  isPressable={false}
                  selectionColor={null}
                >
                  John Doe
                </RCTText>
              </RCTText>
              <RCTText
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                isHighlighted={false}
                selectionColor={null}
              >
                in 
                <RCTText
                  isHighlighted={false}
                  isPressable={false}
                  selectionColor={null}
                >
                  123 Main St
                </RCTText>
              </RCTText>
            </View>
            <View
              className="flex-col gap-3"
            >
              <View
                className="flex-row justify-between"
              >
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  CPF/CNPJ: 
                </RCTText>
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  mocked-taxId
                </RCTText>
              </View>
              <View
                className="flex-row justify-between"
              >
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  Institution: 
                </RCTText>
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  numberOfLines={2}
                  selectionColor={null}
                >
                  Mock Bank
                </RCTText>
              </View>
              <View
                className="flex-row justify-between"
              >
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  Pix Key: 
                </RCTText>
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  mocked-pixkey
                </RCTText>
              </View>
              <View
                className="flex-row justify-between"
              >
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  Identifier: 
                </RCTText>
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  mocked-identifier
                </RCTText>
              </View>
            </View>
          </View>
          <View />
          <RCTText
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            isHighlighted={false}
            selectionColor={null}
          >
            Select the account
          </RCTText>
          <View
            className="p-4 rounded-xl bg-background-50 pb-2"
          >
            <View
              className="flex-row"
            >
              <View
                className="w-1/4"
              >
                <View
                  onLayout={[Function]}
                  style={
                    {
                      "justifyContent": "center",
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessible={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                    testID="select-account"
                  >
                    <Text
                      style={
                        [
                          {
                            "flex": 1,
                            "fontSize": 16,
                            "writingDirection": "ltr",
                          },
                          undefined,
                          {},
                        ]
                      }
                    >
                      Select item
                    </Text>
                    <Image
                      source={1}
                      style={
                        {
                          "height": 20,
                          "tintColor": "gray",
                          "width": 20,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                className="w-3/4"
              >
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                  testID="quote"
                >
                  $ 10.00
                </RCTText>
              </View>
            </View>
            <View
              className="flex-row justify-between"
            >
              <RCTText
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                isHighlighted={false}
                selectionColor={null}
                testID="balance"
              >
                Balance: 
                $ 100.00
              </RCTText>
            </View>
          </View>
          <RCTText
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            isHighlighted={false}
            selectionColor={null}
          >
            The seller will receive the exact value he set. The quantity that will be sent is computed automatically.
          </RCTText>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            dataSet={
              {
                "active": "false",
                "disabled": "false",
                "focus": "false",
                "focusVisible": "false",
                "hover": "false",
              }
            }
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
          >
            <Text
              dataSet={
                {
                  "active": false,
                  "disabled": undefined,
                  "focus": false,
                  "focusVisible": false,
                  "hover": false,
                }
              }
              states={
                {
                  "active": false,
                  "disabled": undefined,
                  "focus": false,
                  "focusVisible": false,
                  "hover": false,
                }
              }
            >
              Pay
               
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollContentView>
  </RCTScrollView>
</View>
`;

exports[`ConfirmPayment component renders correctly for stellar payment 1`] = `
<View
  style={
    [
      {},
      {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    alwaysBounceVertical={true}
    onContentSizeChange={null}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminationRequest={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollShouldSetResponder={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    pagingEnabled={false}
    scrollViewRef={null}
    sendMomentumEvents={false}
    snapToEnd={true}
    snapToStart={true}
    style={
      {
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      }
    }
  >
    <RCTScrollContentView
      collapsable={false}
      style={
        [
          false,
          undefined,
        ]
      }
    >
      <View
        className="flex-1"
        style={
          {
            "paddingTop": 0,
          }
        }
      >
        <View
          className="flex-col gap-4 p-4"
        >
          <View
            className="flex-row justify-between"
          >
            <RCTText
              accessibilityRole="header"
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              isHighlighted={false}
              selectionColor={null}
              style={
                {
                  "fontSize": 37.44,
                  "fontWeight": "bold",
                  "marginVertical": 32,
                }
              }
            >
              Review the payment
            </RCTText>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              dataSet={
                {
                  "active": "false",
                  "focus": "false",
                  "focusVisible": "false",
                  "hover": "false",
                }
              }
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              states={
                {
                  "active": false,
                  "focus": false,
                  "focusVisible": false,
                  "hover": false,
                }
              }
              testID="close-button"
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                meetOrSlice={0}
                minX={0}
                minY={0}
                role="img"
                stroke="currentColor"
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGPath
                      d="M18 6L6 18"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    />
                    <RNSVGPath
                      d="M6 6L18 18"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
          <View
            className="flex-row items-center"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              dataSet={
                {
                  "active": "false",
                  "disabled": "false",
                  "focus": "false",
                  "focusVisible": "false",
                  "hover": "false",
                }
              }
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <RCTText
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                isHighlighted={false}
                selectionColor={null}
                testID="amount"
              >
                $ 10.00
              </RCTText>
            </View>
          </View>
          <View
            className="flex-col gap-7"
          >
            <View>
              <RCTText
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                isHighlighted={false}
                numberOfLines={1}
                selectionColor={null}
              >
                for
                 
                <RCTText
                  isHighlighted={false}
                  isPressable={false}
                  selectionColor={null}
                >
                  John Doe
                </RCTText>
              </RCTText>
              <RCTText
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                isHighlighted={false}
                selectionColor={null}
              >
                in 
                <RCTText
                  isHighlighted={false}
                  isPressable={false}
                  selectionColor={null}
                >
                  123 Main St
                </RCTText>
              </RCTText>
            </View>
            <View
              className="flex-col gap-3"
            >
              <View
                className="flex-row justify-between"
              >
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  Institution:
                </RCTText>
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  Stellar Network
                </RCTText>
              </View>
              <View
                className="flex-row justify-between"
              >
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  Wallet Key:
                </RCTText>
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                >
                  mocke...etkey
                </RCTText>
              </View>
            </View>
          </View>
          <View />
          <RCTText
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            isHighlighted={false}
            selectionColor={null}
          >
            Select the account
          </RCTText>
          <View
            className="p-4 rounded-xl bg-background-50 pb-2"
          >
            <View
              className="flex-row"
            >
              <View
                className="w-1/4"
              >
                <View
                  onLayout={[Function]}
                  style={
                    {
                      "justifyContent": "center",
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessible={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                    testID="select-account"
                  >
                    <Text
                      style={
                        [
                          {
                            "flex": 1,
                            "fontSize": 16,
                            "writingDirection": "ltr",
                          },
                          undefined,
                          {},
                        ]
                      }
                    >
                      Select item
                    </Text>
                    <Image
                      source={1}
                      style={
                        {
                          "height": 20,
                          "tintColor": "gray",
                          "width": 20,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                className="w-3/4"
              >
                <RCTText
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  isHighlighted={false}
                  selectionColor={null}
                  testID="quote"
                >
                  $ 10.00
                </RCTText>
              </View>
            </View>
            <View
              className="flex-row justify-between"
            >
              <RCTText
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                isHighlighted={false}
                selectionColor={null}
                testID="balance"
              >
                Balance: 
                $ 100.00
              </RCTText>
            </View>
          </View>
          <RCTText
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            isHighlighted={false}
            selectionColor={null}
          >
            The seller will receive the exact value he set. The quantity that will be sent is computed automatically.
          </RCTText>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            dataSet={
              {
                "active": "false",
                "disabled": "false",
                "focus": "false",
                "focusVisible": "false",
                "hover": "false",
              }
            }
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
          >
            <Text
              dataSet={
                {
                  "active": false,
                  "disabled": undefined,
                  "focus": false,
                  "focusVisible": false,
                  "hover": false,
                }
              }
              states={
                {
                  "active": false,
                  "disabled": undefined,
                  "focus": false,
                  "focusVisible": false,
                  "hover": false,
                }
              }
            >
              Pay
               
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollContentView>
  </RCTScrollView>
</View>
`;
